src/NetMQ/AsyncReceiveExtensions.cs

```diff
@@ -0,0 +1,219 @@
+        }
+
+
+        /// <summary>
+        /// Receive a single frame from <paramref name="socket"/>, asynchronously, and decode as a string using <paramref name="encoding"/>.
+        /// </summary>
+        /// <param name="socket">The socket to receive from.</param>
+        /// <param name="encoding">The encoding used to convert the frame's data to a string.</param>
+        /// <returns>The content of the received message frame as a string and boolean indicate if another frame of the same message follows..</returns>
+        [NotNull]
+        public static Task<(string, bool)> ReceiveFrameStringAsync([NotNull] this NetMQSocket socket, [NotNull] Encoding encoding)
+        {
+            if (NetMQRuntime.Current == null)
+                throw new InvalidOperationException("NetMQRuntime must be created before calling async functions");
+
+            socket.AttachToRuntime();
+
+            var msg = new Msg();
+            msg.InitEmpty();
+
+            if (socket.TryReceive(ref msg, TimeSpan.Zero))
+            {
+                var str = msg.Size > 0
+                    ? encoding.GetString(msg.Data, msg.Offset, msg.Size)
+                    : string.Empty;
+
+                msg.Close();
+                return Task.FromResult((str, msg.HasMore));
+            }
+
+            TaskCompletionSource<(string, bool)> source = new TaskCompletionSource<(string,bool)>();
+
+            void Listener(object sender, NetMQSocketEventArgs args)
+            {
+                if (socket.TryReceive(ref msg, TimeSpan.Zero))
+                {
+                    var str = msg.Size > 0
+                        ? encoding.GetString(msg.Data, msg.Offset, msg.Size)
+                        : string.Empty;
+                    bool more = msg.HasMore;
+
+                    msg.Close();
+                    socket.ReceiveReady -=  Listener;
+                    source.SetResult((str, more));
+                }
+            }
+
+            socket.ReceiveReady += Listener;
+
+            return source.Task;
+        }
+
+        #endregion
+
+        #region Skipping a message
+
+        /// <summary>
+        /// Receive a single frame from <paramref name="socket"/>, asynchronously, then ignore its content.
+        /// </summary>
+        /// <param name="socket">The socket to receive from.</param>
+        /// <returns>Boolean indicate if another frame of the same message follows</returns>
+        public static Task<bool> SkipFrameAsync([NotNull] this NetMQSocket socket)
+        {
+            if (NetMQRuntime.Current == null)
+                throw new InvalidOperationException("NetMQRuntime must be created before calling async functions");
+
+            socket.AttachToRuntime();
+
+            var msg = new Msg();
+            msg.InitEmpty();
+
+            if (socket.TryReceive(ref msg, TimeSpan.Zero))
+            {
+                bool more = msg.HasMore;
+                msg.Close();
+                return Task.FromResult(more);
```

Comment:
Should probably cache a true result and a false result here to avoid allocating.

Reply:
addressed in https://github.com/zeromq/netmq/pull/804